src/relay/op/nn/nn.cc

```diff
@@ -259,10 +260,25 @@ bool DensePackRel(const Array<Type>& types, int num_inputs, const Attrs& attrs,
   ICHECK(param != nullptr);
 
   ICHECK_EQ(data->shape.size(), 2) << "Only 2D data is supported";
-  ICHECK(weight->shape.size() == 3 || weight->shape.size() == 4) << "Expect weight to be 3D or 4D";
+  ICHECK(weight->shape.size() == 2 || weight->shape.size() == 3 || weight->shape.size() == 4) << "Expect weight to be 2D, 3D or 4D";
 
   Array<tvm::PrimExpr> oshape = data->shape;
-  oshape.Set(1, weight->shape[0] * weight->shape[2]);
+
+  std::string weight_layout = param->weight_layout;
+  std::regex blk_cn_fmt("NC[[:digit:]]+c[[:digit:]]+n");
+  std::regex blk_nc_fmt("NC[[:digit:]]+n[[:digit:]]+c");
+
+  if (weight->shape.size() == 3) {
+    oshape.Set(1, weight->shape[0] * weight->shape[2]);
+  } else if (weight->shape.size() == 4) {
+    if (std::regex_match(weight_layout, blk_cn_fmt)) {
+      oshape.Set(1, weight->shape[0] * weight->shape[3]);
+    } else if (std::regex_match(weight_layout, blk_nc_fmt)) {
+      oshape.Set(1, weight->shape[0] * weight->shape[2]);
+    }
```

Comment:
The same situation as previous with weight rank 3. Layout can be "CN_c_n" or "CN_n_c". With current implementation it will be just skipped without any warning/error message. Could you please improve it?

Reply:
Sure. Fix it.